### DISCUSSION TITLE
Enchance coverage with procedure level data

### DISCUSSION THREAD

**Discussion by jgebal** [utPLSQL/utPLSQL#1329] [Feature Requests]
If package was complied eith plscope enabled we can provide procedure level coverage with cobertura report.
Query to retrieve procrdure start/end lines:
```sql
with
   base as (
      select i.owner, 
             i.object_type, 
             i.object_name, 
             i.name as procedure_name, 
             i.line as start_line, 
             lead(i.line-1 respect nulls) over(order by i.line, i.col) as end_line,
             max(
                case 
                   when i.usage = 'DECLARATION' then 
                      1 
                   else 0 
                end
             ) over (partition by i.signature) as is_private             
        from dba_identifiers i
       where i.object_type = 'PACKAGE BODY'
         and i.type in ('FUNCITON', 'PROCEDURE')
         and i.usage in ('DECLARATION','DEFINITION')
   )
select owner, object_type, object_name, procedure_name, start_line, end_line
  from base
 where is_private = 0;
```

**Comment by lwasylow** [comment]
That would be a great improvment however quite a substantial changes would have to be done to html format so we can provide a quick links to procedure line and some package level summary ( drill into) 
e.g.
PACKAGE1 - 60%
 -> PROCEDURE1 100%
 -> PROCEDURE2 10%

**Comment by jgebal** [comment]
There are 2 aspects to it.
1. make it possible (coverage code changes)
2. make it reportable (reporters code changes)

I would start with No. 1 
We will need to make sure it is accurate.
